### PR TITLE
pr3-retry-failover: isolated voting-ui slice

### PR DESCRIPTION
**Pull request by p0mvn**
## Summary
- Add shared voting retry primitives and centralized transient-error classification used across voting config/service layers.
- Harden vote-server failover by propagating full server sets through voting API call paths instead of single-endpoint wiring.
- Align config refresh retry behavior with the shared policy so transient statuses (including 429/500/502/503/504) retry consistently.
- Fix failover edge cases by removing unsafe `late final` URI reassignment in fallback closures and ensuring PIR transient non-200 responses are retried inside the retry wrapper.

## Test plan
- [x] `flutter test test/services/voting/voting_retry_test.dart`
- [x] `flutter test test/services/voting/voting_api_client_test.dart test/services/voting/pir_snapshot_resolver_test.dart`
- [x] `flutter test test/providers/voting/voting_config_provider_test.dart test/providers/voting/voting_providers_test.dart`
- [x] `flutter analyze lib/src/providers/voting lib/src/services/voting lib/src/features/voting/screens/voting_results_screen.dart`

### CODE REVIEW

**Comment by p0mvn** [comment]
@codex review

File: lib/src/providers/voting/voting_service_providers.dart [hunk 1]
```diff
@@ -38,20 +40,92 @@ final votingConfigLoaderProvider = Provider<VotingConfigLoader>((ref) {
...
+final votingApiReadRetryPolicyProvider = Provider<VotingRetryPolicy>((ref) {
+  return ref.watch(votingTransportRetryPolicyProvider);
+});
+
+/// REST client for chain-facing vote server endpoints.
+final votingApiClientProvider =
+    Provider.family<VotingApiClient, VotingApiServerSet>((ref, servers) {
+      return VotingApiClient(
+        baseUrl: servers.primary,
+        fallbackBaseUrls: servers.failovers,
```
[the remark this answers is not in the document]
Comment (p0mvn) [reply lib/src/providers/voting/voting_service_providers.dart:119]: Addressed in https://github.com/chainapsis/vizor-wallet/commit/7b1cf0cade6f060b8dba074ac61f5088f4b95988 by switching remaining voting callsites to `votingApiClientProvider(config.apiServers)` so failover endpoints are used end-to-end.

File: lib/src/providers/voting/voting_config_provider.dart [hunk 2]
```diff
@@ -66,6 +98,12 @@ class VotingConfigNotifier extends AsyncNotifier<ResolvedVotingConfig> {
       state = AsyncData(config);
     } catch (error, stackTrace) {
       if (!_isCurrentLoad(generation)) return;
+      _recordRefreshFailure(error: error, stackTrace: stackTrace);
+      final lastGoodConfig = _previousResolvedConfig;
+      if (_isRetryableRefreshError(error) && lastGoodConfig != null) {
+        state = AsyncData(lastGoodConfig);
```
[the remark this answers is not in the document]
Comment (p0mvn) [reply lib/src/providers/voting/voting_config_provider.dart:108]: Addressed in https://github.com/chainapsis/vizor-wallet/commit/7b1cf0cade6f060b8dba074ac61f5088f4b95988 by scoping `_previousResolvedConfig` reuse to the active source URL; source switches now fail closed instead of replaying stale config data.

File: lib/src/providers/voting/voting_config_provider.dart [hunk 3]
```diff
@@ -132,12 +184,36 @@ class VotingConfigNotifier extends AsyncNotifier<ResolvedVotingConfig> {
...
+  }
+
+  bool _isRetryableRefreshError(Object error) {
+    if (error is TimeoutException ||
+        error is SocketException ||
+        error is HttpException) {
+      return true;
+    }
+    if (error is VotingHttpException) {
+      return error.statusCode == 502 || error.statusCode == 503;
```
[the remark this answers is not in the document]
Comment (p0mvn) [reply lib/src/providers/voting/voting_config_provider.dart]: Already addressed earlier in https://github.com/chainapsis/vizor-wallet/commit/4ef540a4d658c8917c6c4983995eb02212a640b8, which routes config retryability through `isRetryableVotingError` so transient 429/500/502/503/504 statuses remain retryable.

File: lib/src/services/voting/voting_api_client.dart [hunk 4]
```diff
@@ -60,10 +91,24 @@ class VotingApiClient {
 
   /// Fetches the currently active round, returning null when the chain has none.
```
[the remark this answers is not in the document]
Comment (p0mvn) [reply lib/src/services/voting/voting_api_client.dart:92]: Addressed in https://github.com/chainapsis/vizor-wallet/commit/706a8810baa46d7f9e29168a51ce8f388881f422 by removing `late final` URI reassignment in failover closures for `getActiveRoundStatus()` and `getTxConfirmation()`.

File: lib/src/services/voting/pir_snapshot_resolver.dart [hunk 5]
```diff
@@ -130,9 +142,10 @@ class PirSnapshotResolver {
...
   }) async {
     try {
-      final response = await _httpClient.get(
-        _rootUri(endpoint),
-        timeout: _timeout,
+      final response = await withVotingRetry(
+        policy: _retryPolicy,
+        delay: _delay,
+        operation: () => _httpClient.get(_rootUri(endpoint), timeout: _timeout),
       );
```
[the remark this answers is not in the document]
Comment (p0mvn) [reply lib/src/services/voting/pir_snapshot_resolver.dart:164]: Addressed in https://github.com/chainapsis/vizor-wallet/commit/706a8810baa46d7f9e29168a51ce8f388881f422 by throwing retryable PIR non-200 responses inside `withVotingRetry`, so transient `/root` gateway failures are retried before diagnostics are emitted.

File: lib/src/providers/voting/voting_tree_sync_provider.dart [hunk 6]
```diff
@@ -59,7 +60,7 @@ class VotingTreePreSyncService {
         dbPath: dbPath,
         accountUuid: accountUuid,
         roundId: roundId,
-        nodeUrl: config.apiBaseUrl.toString(),
+        nodeUrl: primaryApiServer.toString(),
```
[the remark this answers is not in the document]
Comment (p0mvn) [reply lib/src/providers/voting/voting_tree_sync_provider.dart:63]: Tracking separately
